src/main/java/focusedCrawler/crawler/cookies/OkHttpCookieJar.java

```diff
@@ -43,6 +42,30 @@ public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
         return validCookies;
     }
 
+    @Override
+    public List<Cookie> loadForRequest(HttpUrl url) {
+        List<Cookie> allCookies = new ArrayList<>();
+
+        if(!url.host().equals(url.topPrivateDomain()) && !url.host().equals("www."+url.topPrivateDomain())) {
+            String strUrl = url.host();
+            for (int i = 0; i < strUrl.length(); i++) {
+
+                if (strUrl.charAt(i) == '.') {
+                    HttpUrl url1 = HttpUrl.parse("https://"+strUrl.substring(i + 1));
+
+                    allCookies.addAll(loadValidCookies(url1));
+
+                    if (strUrl.substring(i + 1).equals(url.topPrivateDomain())) {
```

Comment:
I don't know how expensive `url.topPrivateDomain()` is , so it is better to store the value in a local variable.